src/config.rs

```diff
@@ -344,59 +333,53 @@ pub fn get_config_dir() -> Result<(PathBuf, PathSource), TealdeerError> {
     // Otherwise, fall back to the user config directory.
     match get_app_root(AppDataType::UserConfig, &crate::APP_INFO) {
         Ok(dirs) => Ok((dirs, PathSource::OsConvention)),
-        Err(_) => Err(ConfigError(
-            "Could not determine the user config directory.".into(),
-        )),
+        Err(_) => Err(anyhow!("Failed to determine the user config directory")),
     }
```

Comment:
Yeah, probably makes sense.

Reply:
Ah, no, actually this doesn't work because `AppDirsError` doesn't seem to implement `Error`. However, we can fake it:

```diff
-        Err(_) => Err(anyhow!("Failed to determine the user config directory")),
+        Err(e) => Err(anyhow!("{}", e).context("Failed to determine the user config directory")),
```

Reply:
Ah, never mind, I'm stupid 😄

Reply:
Are you sure? I can do 
```rust
    let path = get_app_root(AppDataType::UserConfig, &crate::APP_INFO)
        .context("Failed to determine the user config directory")?;
    Ok((path, PathSource::OsConvention))
```
on my machine and we also do it [in another place](https://github.com/dbrgn/tealdeer/pull/249#discussion_r780711734)

Reply:
Yeah, I tried to call `context` on the inner `e`, that's why it didn't work. It's fixed now.